collections/misc/collbackup.php

```diff
@@ -1,25 +1,29 @@
 <?php
 include_once('../../config/symbini.php');
-include_once($SERVER_ROOT.'/content/lang/collections/misc/collbackup.'.$LANG_TAG.'.php');
-header("Content-Type: text/html; charset=".$CHARSET);
+if($LANG_TAG != 'en' && file_exists($SERVER_ROOT . '/content/lang/collections/misc/collbackup.' . $LANG_TAG . '.php')){
+	include_once($SERVER_ROOT . '/content/lang/collections/misc/collbackup.' . $LANG_TAG . '.php');
+}
+else include_once($SERVER_ROOT . '/content/lang/collections/misc/collbackup.en.php');
+header('Content-Type: text/html; charset=' . $CHARSET);
 
-$collid = array_key_exists("collid",$_REQUEST)?$_REQUEST["collid"]:0;
-$action = array_key_exists("formsubmit",$_REQUEST)?$_REQUEST["formsubmit"]:'';
-$cSet = array_key_exists("cset",$_REQUEST)?$_REQUEST["cset"]:'';
+$collid = isset($_REQUEST['collid']) ? filter_var($_REQUEST['collid'], FILTER_SANITIZE_NUMBER_INT) : 0;
+$action = isset($_REQUEST['formsubmit']) ? $_REQUEST['formsubmit'] : '';
```

Comment:
The pattern we've been using in the Checkmarx-test branch so far has been running `htmlspecialchars($foo, HTML_SPECIAL_CHARS_FLAGS)` in these lines, rather than in the global sanitize (`cleanOutStr`, was it?). @GregPost-ASU , Nikita, and I met about it briefly, and we're thinking, "let's just give checkmarx what it wants. It's just a much work to go through and dismiss all of the false positives on checkmarx." What do you think?

Reply:
I don't think Checkmarx is gonna be smart enough to distinguish an `input` of `type="hidden"` from one that is not hidden. Is there any harm if we implement the sanitizing?